3rdParty/fuerte/src/AsioSockets.h

```diff
@@ -125,18 +125,22 @@ struct Socket<fuerte::SocketType::Ssl> {
             return;
           }
 
-          // Perform SSL handshake and verify the remote host's certificate.
-          socket.next_layer().set_option(asio_ns::ip::tcp::no_delay(true));
-          if (verify) {
-            socket.set_verify_mode(asio_ns::ssl::verify_peer);
-            socket.set_verify_callback(
-                asio_ns::ssl::rfc2818_verification(config._host));
-          } else {
-            socket.set_verify_mode(asio_ns::ssl::verify_none);
+          try {
+            // Perform SSL handshake and verify the remote host's certificate.
+            socket.next_layer().set_option(asio_ns::ip::tcp::no_delay(true));
+            if (verify) {
+              socket.set_verify_mode(asio_ns::ssl::verify_peer);
+              socket.set_verify_callback(
+                  asio_ns::ssl::rfc2818_verification(config._host));
+            } else {
+              socket.set_verify_mode(asio_ns::ssl::verify_none);
+            }
+            socket.async_handshake(asio_ns::ssl::stream_base::client,
+                                   std::move(done));
+            
+          } catch(boost::system::system_error const& exc) {
+            done(exc.code());
```

Comment:
Is it ok to call `done(...)` here if it was moved-from in line 138? I think line 138 must not use `std::move(done)` but just `done`.

Reply:
AFAIK invoking `operator()` on a moved-from function object will throw an exception, so the catch block itself does not look safe to me.